### PR TITLE
Always perform custom values merge

### PR DESCRIPTION
**Pull request by sjmiller609**
There was an issue with the 0.15.0 prod cloud release where the pgbouncer logging values were not set: https://github.com/astronomer/airflow-chart/blob/c6b1288ceff874948aa2f23aef189d290b0f13d7/values.yaml#L303-L305

Since these were nil, pgbouncer failed to start with error ~ 'invalid config'. This PR includes a new test case that reproduces this upgrade issue.

I had an invalid assumption that when these were missing from the user-provided values during a helm upgrade, that a the classic nil pointer exception would occur, then proceed into this "if" block https://github.com/astronomer/commander/blob/b2c9399dcc0e51b6dbf75d4fd06b499f69b7a94c/helm/client.go#L175

In practice, and reproduced in this new test case, we have found that nil pointer exception was not triggered, and so we need to always perform our full values merging as we do in the "if" block.

We discussed this option during development, and what we don't like about it is that there is a race condition between fetching the existing values and setting the new ones. However, I think it is the only known option to pass the next test case. In this patch, we remove the "if" block, and instead by default always three way merge the new chart default values, existing user-provided values, and the new user-provided values.

I will separately PR into master branch, linking to this PR, rather than cherry picking in this case. that is because this patch is needed to release 0.15.0 to enterprise.

### CODE REVIEW

File: helm/client.go [hunk 1]
```diff
@@ -171,53 +172,34 @@ func (c *Client) UpdateRelease(releaseName, chartName, chartVersion, namespace s
...
-		rel, err := client.Run(releaseName, ch, mergedOptions)
-		if err != nil {
-			logger.Errorf("error while trying to upgrade: %v", err)
-			return nil, err
-		}
-		logger.Infof("the upgrade worked after retrying with all values for release: %s", releaseName)
-		return rel, nil
+	// In the Coalesce command, the first argument takes priority
+	// merge in chart default values, imitating --reset-values
+	allValues := chartutil.CoalesceTables(existingValues, ch.Values)
```
Comment (schnie) [inline helm/client.go:192]: @sjmiller609 so does `ch.Values` contain all the default values that are baked into the chart? This line is merging the `existingValues` _on top_ of the baked in defaults correct?

Comment (andriisoldatenko) [reply helm/client.go:192]: this code is just copy paste from internal helm get values impl:
https://github.com/helm/helm/blob/8ea6b970ecd02365a230420692350057d48278e5/pkg/action/get_values.go#L53

Comment (sjmiller609) [reply helm/client.go:192]: that's right greg

Comment (schnie) [reply helm/client.go:192]: Ah, got it. Is my description of the behavior correct though?

**Comment by andriisoldatenko** [comment]
@schnie i proposed to add test each release e.g. to emulate upgrade on top.

**Comment by sjmiller609** [comment]
> This is awesome, should help us avoid a classic problem we've hit all too often.
> 
> One general question. We have some specific chart versions specified in our tests, like 0.15.2. How do these carry forward? Would we update these versions or just leave them how they are?

that's a good question. at least for regressions, they should stay the same. but for other cases, it might make sense to fetch the new chart versions. However, this makes the tests less reproducible, and one could argue that's testing airflow chart rather than commander itself. I'm not sure about it, but I feel that maybe we only worry about new versions when there is some new issue and we make another test, assuming that the cases we are concerned about from commander's perspective are covered under the current suite.

File: helm/client.go (hunk 1 above)
Comment (sjmiller609) [reply helm/client.go:192]: yes it is: https://github.com/helm/helm/blob/8ea6b970ecd02365a230420692350057d48278e5/pkg/chartutil/coalesce.go#L174-L203

**Comment by sjmiller609** [comment]
I did a merge commit on accident, so I did a squash the force push - as a result the hash above is not in the release branch, the hash is 9c06de7

**Pull request merged**